src/main/java/com/dabsquared/gitlabjenkins/connection/GitLabConnection.java

```diff
@@ -277,28 +234,23 @@ public FormValidation doCheckReadTimeout(@QueryParameter Integer value) {
         public FormValidation doTestConnection(
                 @QueryParameter String url,
                 @QueryParameter String apiTokenId,
-                @QueryParameter String clientBuilderId,
                 @QueryParameter boolean ignoreCertificateErrors,
                 @QueryParameter int connectionTimeout,
                 @QueryParameter int readTimeout) {
             Jenkins.get().checkPermission(Jenkins.ADMINISTER);
             try {
-                new GitLabConnection(
-                                "",
-                                url,
-                                apiTokenId,
-                                clientBuilderId,
-                                ignoreCertificateErrors,
-                                connectionTimeout,
-                                readTimeout)
-                        .getClient(null, null)
```

Comment:
As described above, we cannot break compatibility nor is it desirable to drop support for autodetection. This should be reverted.